### PR TITLE
Handle cases when connection is failed for keycloak related requests

### PR DESCRIPTION
**Pull request by RomanNikitenko**
### What does this PR do?
Do not wrap UnauthorizedException in ServerException at connecting to GitHub API using OAuth

### What issues does this PR fix or reference?
#8288 

Signed-off-by: Roman Nikitenko <rnikiten@redhat.com>

### CODE REVIEW

**Comment by RomanNikitenko** [comment]
ci-test

**Review by vparfonov — APPROVED** [review]

**Comment by RomanNikitenko** [comment]
ci-test

File: core/che-core-api-core/src/main/java/org/eclipse/che/api/core/util/HttpJsonRequestUtil.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+import com.google.gson.JsonSyntaxException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.eclipse.che.commons.annotation.Nullable;
+
+/**
+ * Provides useful methods to simplify work with {@link
+ * org.eclipse.che.api.core.rest.HttpJsonRequest} instances.
+ *
+ * @author Roman Nikitenko
```
Comment (skabashnyuk) [inline core/che-core-api-core/src/main/java/org/eclipse/che/api/core/util/HttpJsonRequestUtil.java]: Look like specific to KeycloakOAuthAuthenticationService.java class. Plese move it close to service

**Review by skabashnyuk — APPROVED** [review]

File: core/che-core-api-core/src/main/java/org/eclipse/che/api/core/util/HttpJsonRequestUtil.java (hunk 1 above)
Comment (RomanNikitenko) [reply core/che-core-api-core/src/main/java/org/eclipse/che/api/core/util/HttpJsonRequestUtil.java]: @skabashnyuk 
In DefaultHttpJsonRequest class we create formatted error message with corresponding info when request is failed.
We should have ability to extract this info from formatted message. 
At the moment I use this class to extract and analyze code and error message of response in KeycloakOAuthAuthenticationService.java class only.
But we can face the need to extract formatted in DefaultHttpJsonRequest class info in another services.
So I think this util class more related to DefaultHttpJsonRequest class, but up to you.
WDYT?

**Comment by RomanNikitenko** [comment]
@vparfonov @skabashnyuk I've updated the PR, please review.

**Pull request closed without merging**